graylog2-server/src/main/java/org/graylog2/web/resources/AppConfigResource.java

```diff
@@ -68,7 +78,19 @@ public String get(@Context HttpHeaders headers) {
         final Map<String, Object> model = ImmutableMap.of(
             "rootTimeZone", configuration.getRootTimeZone(),
             "serverUri", baseUri.resolve(HttpConfiguration.PATH_API),
-            "appPathPrefix", baseUri.getPath());
+            "appPathPrefix", baseUri.getPath(),
+            "pluginUISettings", buildPluginUISettings());
         return templateEngine.transform(template, model);
     }
+
+    private String buildPluginUISettings() {
+        Map<String, Object> pluginUISettings = settingsProviders.entrySet().stream().collect(Collectors.toMap(
+                Map.Entry::getKey,
+                entry -> entry.getValue().pluginSettings()));
+        try {
+            return objectMapper.writeValueAsString(pluginUISettings);
```

Comment:
For debugging purposes, it would be nice to pretty-print this. We can do it using `ObjectWriter#withPrettyPrinter`.

Reply:
Here's what `config.js` looks like with the update in place:

![Screen Shot 2021-03-01 at 11 33 24 AM](https://user-images.githubusercontent.com/6466251/109527696-f1bd0a80-7a81-11eb-9e89-5b38138ad8dd.png)

I'll push a new commit shortly